Tests/ttx/ttx_test.py

```diff
@@ -263,5 +267,569 @@ def test_guessFileType_invalid_path(self):
+    options.listTables = True
+    ttx.ttList(inpath, fakeoutpath, options)
+    out, err = capsys.readouterr()
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "hmtx", "fpgm", "prep", "cvt ", "loca", "glyf", "post", "gasp", "DSIG")
+    # confirm that expected tables are printed to stdout
+    for table in expected_tables:
+        assert table in out
+    # test for one of the expected tag/checksum/length/offset strings
+    assert "OS/2  0x67230FF8        96       376" in out
+
+
+def test_ttx_ttlist_otf(capsys, tmpdir):
+    inpath = os.path.join("Tests", "ttx", "data", "TestOTF.otf")
+    fakeoutpath = os.path.join(str(tmpdir), "TestOTF.ttx")
+    options = ttx.Options([("", "")], 1)
+    options.listTables = True
+    ttx.ttList(inpath, fakeoutpath, options)
+    out, err = capsys.readouterr()
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "post", "CFF ", "hmtx", "DSIG")
+    # confirm that expected tables are printed to stdout
+    for table in expected_tables:
+        assert table in out
+    # test for one of the expected tag/checksum/length/offset strings
+    assert "OS/2  0x67230FF8        96       272" in out
+
+
+def test_ttx_ttlist_woff(capsys, tmpdir):
+    inpath = os.path.join("Tests", "ttx", "data", "TestWOFF.woff")
+    fakeoutpath = os.path.join(str(tmpdir), "TestWOFF.ttx")
+    options = ttx.Options([("", "")], 1)
+    options.listTables = True
+    options.flavor = "woff"
+    ttx.ttList(inpath, fakeoutpath, options)
+    out, err = capsys.readouterr()
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "post", "CFF ", "hmtx", "DSIG")
+    # confirm that expected tables are printed to stdout
+    for table in expected_tables:
+        assert table in out
+    # test for one of the expected tag/checksum/length/offset strings
+    assert "OS/2  0x67230FF8        84       340" in out
+
+
+def test_ttx_ttlist_woff2(capsys, tmpdir):
+    inpath = os.path.join("Tests", "ttx", "data", "TestWOFF2.woff2")
+    fakeoutpath = os.path.join(str(tmpdir), "TestWOFF2.ttx")
+    options = ttx.Options([("", "")], 1)
+    options.listTables = True
+    options.flavor = "woff2"
+    ttx.ttList(inpath, fakeoutpath, options)
+    out, err = capsys.readouterr()
+    expected_tables = ("head", "hhea", "maxp", "OS/2", "name", "cmap", "hmtx", "fpgm", "prep", "cvt ", "loca", "glyf", "post", "gasp")
+    # confirm that expected tables are printed to stdout
+    for table in expected_tables:
+        assert table in out
+    # test for one of the expected tag/checksum/length/offset strings
+    assert "OS/2  0x67230FF8        96         0" in out
+
+# -------------------
+# main function tests
+# -------------------
+
+
+def test_ttx_main_default_ttf_dump_to_ttx(tmpdir):
+    inpath = os.path.join("Tests", "ttx", "data", "TestTTF.ttf")
+    outpath = os.path.join(str(tmpdir), "TestTTF.ttx")
+    args = ["-o", outpath, inpath]
+    ttx.main(args)
+    assert os.path.isfile(outpath)
```

Comment:
http://py.readthedocs.io/en/latest/path.html#checking-path-types

Reply:
On hold until we resolve issue in https://github.com/fonttools/fonttools/pull/1261#discussion_r189875115